dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataSqlQueryFactoryBuilderTest.java

```diff
@@ -0,0 +1,147 @@
+        "SELECT "
+            + "L.CollectTimeStamp, L.UserName, L.StatementType, L.AppID, L.DefaultDatabase, "
+            + "L.ErrorCode, L.ErrorText, L.FirstRespTime, L.LastRespTime, L.NumResultRows, "
+            + "L.QueryText, L.ReqPhysIO, L.ReqPhysIOKB, L.RequestMode, L.SessionID, L.SessionWDID, "
+            + "L.Statements, L.TotalIOCount, L.WarningOnly, L.StartTime "
+            + "FROM dbc.DBQLogTbl L "
+            + "WHERE L.StartTime >= CAST('2023-03-04T00:00:00Z' AS TIMESTAMP) "
+            + "AND L.StartTime < CAST('2023-03-09T00:00:00Z' AS TIMESTAMP)";
+
+    //  Act
+    SqlQueryFactory factory =
+        TeradataSqlQueryFactoryBuilder.startBuildingFrom(arguments).within(interval).build();
+    String generatedQuery = factory.getSql((c, t) -> true);
+
+    //  Assert
+    assertEquals(expectedQuery, generatedQuery);
+  }
+
+  @Test
+  public void test_buildStandardFactory() throws IOException, MetadataDumperUsageException {
+    //  Arrange
+    ConnectorArguments arguments = new ConnectorArguments(new String[] {"--connector", "teradata"});
+    ZonedDateTime endDate = ZonedDateTime.parse("2023-03-09T00:00:00Z");
+    ZonedDateTime startDate = endDate.minusDays(5).truncatedTo(ChronoUnit.DAYS);
+    ZonedInterval interval = new ZonedInterval(startDate, endDate);
+    String expectedQuery =
+        "SELECT "
+            + "L.QueryID, ST.SQLRowNo, ST.SQLTextInfo, L.UserName, L.CollectTimeStamp, "
+            + "L.StatementType, L.AppID, L.DefaultDatabase, L.ErrorCode, L.ErrorText, "
+            + "L.FirstRespTime, L.LastRespTime, L.NumResultRows, L.QueryText, L.ReqPhysIO, "
+            + "L.ReqPhysIOKB, L.RequestMode, L.SessionID, L.SessionWDID, L.Statements, "
+            + "L.TotalIOCount, L.WarningOnly, L.StartTime "
+            + "FROM dbc.DBQLogTbl L "
+            + "LEFT OUTER JOIN dbc.DBQLSQLTbl ST ON L.QueryID=ST.QueryID "
+            + "WHERE L.ErrorCode=0 AND L.StartTime >= CAST('2023-03-04T00:00:00Z' AS TIMESTAMP) "
+            + "AND L.StartTime < CAST('2023-03-09T00:00:00Z' AS TIMESTAMP) "
+            + "AND L.UserName <> 'DBC'";
+
+    //  Act
+    SqlQueryFactory factory =
+        TeradataSqlQueryFactoryBuilder.startBuildingFrom(arguments).within(interval).build();
+    String generatedQuery = factory.getSql((c, t) -> true);
+
+    //  Assert
+    assertEquals(expectedQuery, generatedQuery);
+  }
+
+  @Test
+  public void test_buildAssessmentFactory() throws IOException, MetadataDumperUsageException {
+    //  Arrange
+    ConnectorArguments arguments =
+        new ConnectorArguments(new String[] {"--connector", "teradata", "--assessment"});
+    ZonedDateTime endDate = ZonedDateTime.parse("2023-03-09T00:00:00Z");
+    ZonedDateTime startDate = endDate.minusDays(5).truncatedTo(ChronoUnit.DAYS);
+    ZonedInterval interval = new ZonedInterval(startDate, endDate);
+    String expectedQuery =
+        "SELECT L.QueryID, ST.SQLRowNo, ST.SQLTextInfo, L.AbortFlag, L.AcctString,"
```

Comment:
I would suggest to put the test queries in a more readable form in a plain text `*.sql` file in the `src/test/resources` directory. One query per file.

Compare:
https://github.com/google/dwh-migration-tools/blob/858e7988f048024eeb9593adaf5d84a395c2c1e0/dumper/app/src/test/resources/dumper-test/oracle.sql